internal/runners/push/push.go

```diff
@@ -209,12 +209,24 @@ func (r *Push) Run(params PushParams) error {
 	}
 
 	// Update the project at the given commit id.
-	err = model.UpdateProjectBranchCommitWithModel(targetPjm, branch.Label, commitID)
-	if err != nil {
-		if errs.Matches(err, &model.ErrUpdateBranchAuth{}) {
-			return locale.WrapInputError(err, "push_project_branch_no_permission", "You do not have permission to push to {{.V0}}.", targetNamespace.String())
-		} else {
-			return locale.WrapError(err, "push_project_branch_commit_err", "Failed to update new project {{.V0}} to current commitID.", targetNamespace.String())
+	// Currently, the build planner doesn't support attaching a staged commit
+	// without a parent commit ID
+	if branch.CommitID != nil {
+		bp := model.NewBuildPlannerModel(r.auth)
+		logging.Debug("Attaching staged commit via build planner")
+		err = bp.AttachStagedCommit(targetNamespace.Owner, targetNamespace.Project, branch.CommitID.String(), commitID.String(), branch.Label)
+		if err != nil {
+			return locale.WrapError(err, "err_push_attach_staged_commit", "Failed to attach staged commit to project.")
+		}
```

Comment:
I'm trying to understand what this is doing. If the branch has a commit attached to it, then we can't use `UpdateProjectBranch..`? That seems inverted at least in terms of my understanding reading this code. If that is the logic we want please add comments explaining why.

Additionally, we're not supplying a branch here, so what's getting updated?